Form/DataTransformer/ProviderDataTransformer.php

```diff
@@ -28,13 +29,19 @@ class ProviderDataTransformer implements DataTransformerInterface
     protected $options;
 
     /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
      * @param Pool   $pool
      * @param string $class
      * @param array  $options
      */
-    public function __construct(Pool $pool, $class, array $options = array())
+    public function __construct(Pool $pool, $class, array $options = array(), LoggerInterface $logger = null)
```

Comment:
Could you use the `LoggerAwareTrait` instead and use the `NullLogger` by default?